src/lib/stack-client.js

```diff
@@ -0,0 +1,236 @@
+  UnavailableStackException,
+  UnavailableSettingsException,
+  UnauthorizedStackException
+} from './exceptions'
+
+import getIcon from './icon'
+
+let cozyClient = undefined
+
+const errorStatuses = {
+  '401': UnauthorizedStackException,
+  '403': ForbiddenException,
+  '404': NotFoundException,
+  '405': MethodNotAllowedException,
+  '500': ServerErrorException
+}
+
+// Utilizy to fetch the resources with cozy-client
+// maintening the compatibility with the legacy 
+// standalone cozy-bar client
+const fetchJSON = function(method, path, body, options={}) {
+  
+  // We mirror here a few lines from cozy-stack-client
+  // because we want a customized fetchJSON
+  const headers = (options.headers = options.headers || {})
+  headers['Accept'] = 'application/json'
+  if (method !== 'GET' && method !== 'HEAD' && body !== undefined) {
+    if (!headers['Content-Type']) {
+      headers['Content-Type'] = 'application/json'
+      body = JSON.stringify(body)
+    }
+  }
+
+  cozyClient.getStackClient().fetch(method, path, body, options).then(
+    resp => {
+      if (typeof errorStatuses[resp.status] === 'function') {
+        throw new errorStatuses[resp.status]()
+      }    
+      const contentType = resp.headers.get('content-type')
+      const isJson = contentType && contentType.indexOf('json') >= 0
+      if (!isJson) {
+        throw new Error("Server response not in JSON")
+      }
+      return resp.json()
+    }
+  )
+}
+
+// Cache a request 
+// with a defaultValue in case of 404
+const withCache(fn, defaultValue) {
+  let cache = undefined;
+  const is404 = function(error) {
+    return error.name === 'FetchError' &&
+       error.status && 
+       error.status === 404
+  }
+  return function() {
+    if (cache === undefined) {
+      return fn().then(result => 
+        cache = result
+      ).catch( error => 
+        cache = is404(error) ? defaultValue : undefined
+      ) 
+    } else {
+      return Promise.resolve(cache)
+    }
+  }
+}
+
+// List installed applications
+const getApps = function () {
+  return fetchJSON('GET', '/apps/').then(
+    json => {
+      if (json.error) throw new Error(json.error)
+      else return json.data
+    }
+  )
+}
+
+const getApp = function(slug) {
+  if (!slug) {
+    throw new Error('Missing slug')
+  }
+  return fetchJSON('GET', `/apps/${slug}`).then(
+    json => {
+      if (json.error) throw new Error(json.error)
+      else return json.data
+    }
+  )
+}
+
+
+// Current context
+const getContext = function () {
+  return fetchJSON('GET', '/settings/context')
+)
+
+// default value when no quota is provided
+const defaultQuota = 10**12 // 1 Tera
+
+// Storage/quota usage
+const getStorageData = function() {
+  return fetchJSON('GET', '/settings/disk-usage').then( 
+    usage => { 
+      return {
+        usage: parseInt(json.data.attributes.used, 10),
+        // TODO Better handling when no quota provided
+        quota: parseInt(json.data.attributes.quota, 10) || defaultQuota,
```

Comment:
Maybe just add a `isNaN` check?

Reply:
It was from the legacy code. I fear they expect the number to be in a string

@gregorylegarec ?

Reply:
We do https://github.com/cozy/cozy-stack/blob/master/docs/settings.md#response